valuable/src/valuable.rs

```diff
@@ -3,11 +3,63 @@ use crate::{Slice, Value, Visit};
 use core::fmt;
 use core::num::Wrapping;
 
+/// A type that can be converted to a [`Value`].
+///
+/// `Valuable` types are inspected by defining a [`Visit`] implementation and
+/// using it whewn calling [`Valuable::visit`]. See [`Visit`] documentation for
+/// more details.
+///
+/// The `Valuable` procedural macro makes implementing `Valuable` easy. Users
+/// can add add `#[derive(Valuable)]` to their types.
+///
+/// `Valuable` provides implementations for many Rust primitives and standard
+/// library types.
+///
+/// Types implementing `Valuable` may also implement one of the more specific
+/// traits: [`Structable`], [`Enumerable`], [`Listable`], and [`Mappable`].
```

Comment:
might be nice if this briefly mentioned what those traits are for, maybe something like
```suggestion
/// Types implementing `Valuable` may also implement one of the more specific
/// traits: [`Structable`], [`Enumerable`], [`Listable`], and [`Mappable`]. These traits
/// should be implemented when the type is a nested container of other `Valuable` types.
```

or something like that?